src/cpp/shared/game/event_loop.cpp

```diff
@@ -20,3 +20,24 @@ void __fastcall Process1WindowsMessage() {
 }
 
 
+unsigned long UTickCount() {
+  return KBTickCount();
+}
+
+/*
+ * This is a slightly hacky fix to the bug described in https://www.celestialheavens.com/forum/7/16792 ,
+ * where a signed overflow of GetTickCount() can cause the game to wait for up to 25 days
+ * 
+ * My preferred fix would involve eliminating all calls to DelayTilMilli to use DelayTil instead,
+ * as DelayTilMilli can't be sure that you don't really want to wait for 25 days. However, properly
+ * using unsigned ints here means that, in the event that a caller of DelayTilMilli inadvertently passes
+ * in a very small number, it will never wait for very long.
+ */
+void __fastcall DelayTilMilli(long tick) {
+  unsigned long uTick = tick;
+  while (UTickCount() < uTick)
```

Comment:
What about `static_cast<unsigned long>(KBTickCount())` instead of a new function?  It's ugly, but it makes the key bit (using a big enough unsigned type) obvious.